stdlib/runtime/gc.gr

```diff
@@ -172,11 +283,25 @@ let rec decRef = (userPtr: WasmI32, ignoreZeros: Bool) => {
     //   void
     // }
 
+    if (WasmI32.ne(_DEBUG_HOOK_DECREF, 0n)) {
+      callDebugHookDecRef(userPtr, ignoreZeros)
+      void
+    }
+
     if (WasmI32.eqz(refCount)) {
       if (ignoreZeros) {
         userPtr
       } else {
-        throw DecRefError
+        // Throwing causes last logged print statements to not be printed,
+        // so it becomes very difficult to debug memory problems. Maybe a
+        // buffer is not being flushed? If the debugging hooks are active then
+        // behave ad if ignoreZeros was true to continue executing and give the
+        // debugging hooks an opportunity to report the issue.
+        if (WasmI32.ne(_DEBUG_HOOK_DECREF, 0n)) {
```

Comment:
Very interesting. I wonder if we could get around this by adding a call to the flush syscall in the `if (WasmI32.ne(_DEBUG_HOOK_DECREF, 0n))` expression? 🤔